sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSonarWayProfile.java

```diff
@@ -39,21 +40,40 @@
 @SonarLintSide
 public class JavaSonarWayProfile implements BuiltInQualityProfilesDefinition {
 
+  private final SonarRuntime sonarRuntime;
+
+  public JavaSonarWayProfile(SonarRuntime sonarRuntime) {
+    this.sonarRuntime = sonarRuntime;
+  }
+
   @Override
   public void define(Context context) {
     NewBuiltInQualityProfile sonarWay = context.createBuiltInQualityProfile("Sonar way", Java.KEY);
     sonarWay.activateRule("common-" + Java.KEY, "DuplicatedBlocks");
     Profile jsonProfile = readProfile();
     Map<String, String> keys = legacyKeys();
     for (String key : jsonProfile.ruleKeys) {
-      sonarWay.activateRule(CheckList.REPOSITORY_KEY, keys.get(key));
+      if (shouldActivateRule(key)) {
+        sonarWay.activateRule(CheckList.REPOSITORY_KEY, keys.get(key));
+      }
     }
 
     SonarComponents.getSecurityRuleKeys().forEach(key -> sonarWay.activateRule(CheckList.REPOSITORY_KEY, key));
 
     sonarWay.done();
   }
 
+  private boolean shouldActivateRule(String ruleKey) {
+    JavaRulesDefinition.RuleMetadata ruleMetadata = JavaRulesDefinition.readRuleMetadata(ruleKey);
+    if (ruleMetadata == null) {
```

Comment:
IMHO It would be more readable in one expression
```
return ruleMetadata == null || !ruleMetadata.isSecurityHotspot() || SecurityHotspots.securityHotspotsSupported(sonarRuntime);
```

Reply:
well, I actually prefer the other style :)